File: src/main/java/no/nav/familie/ks/oppslag/dokarkiv/api/ArkiverDokumentRequest.java
package no.nav.familie.ks.oppslag.dokarkiv.api;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import java.time.LocalDateTime;
import java.util.List;

public class ArkiverDokumentRequest {

    @NotBlank
    private String fnr;

    private boolean forsøkFerdigstill;
    @NotEmpty
    private List<Dokument> dokumenter;

    public ArkiverDokumentRequest() {
    }

    public ArkiverDokumentRequest(String fnr, String navn, boolean forsøkFerdigstill, List<Dokument> dokumenter) {
        this.fnr = fnr;
        this.forsøkFerdigstill = forsøkFerdigstill;
        this.dokumenter = dokumenter;
    }

    public String getFnr() {
        return fnr;
    }


    public boolean isForsøkFerdigstill() {
        return forsøkFerdigstill;
    }

    public List<Dokument> getDokumenter() {
        return dokumenter;
    }
}

File: src/main/java/no/nav/familie/integrasjoner/dokarkiv/api/ArkiverDokumentRequest.kt
package no.nav.familie.integrasjoner.dokarkiv.api

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotEmpty


data class ArkiverDokumentRequest(@field:NotBlank val fnr: String,
                                  val forsøkFerdigstill: Boolean,
                                  @field:NotEmpty val dokumenter: List<Dokument>,
                                  val fagsakId: String?= null,
                                  val journalførendeEnhet: String? = null) //Skal ikke settes for innkommende hvis Ruting gjøres av BRUT001
